runtime/interpreter/encode.go

```diff
@@ -543,11 +581,28 @@ func (e *Encoder) prepareString(v *StringValue) string {
 	return v.Str
 }
 
-func (e *Encoder) prepareArray(v *ArrayValue) ([]interface{}, error) {
+// joinPath returns the path for a nested item, for example the index of an array,
+// the key of a dictionary, or the field name of a composite.
+//
+// \x1F = Information Separator One
+//
+func joinPath(elements []string) string {
+	return strings.Join(elements, "\x1F")
```

Comment:
Would it be possible for someone to pass in a path that contains this hex sequence?

Reply:
Currently it is not possible to, because we only allow `[a-zA-Z0-9_]` in identifiers. But good point, if we would add support for converting arbitrary strings to paths / path identifiers, we will need to reject this character.